cmd/internal/find/walker.go

```diff
@@ -0,0 +1,59 @@
+package find
+
+import (
+	"fmt"
+
+	"github.com/puppetlabs/wash/api/client"
+	"github.com/puppetlabs/wash/cmd/internal/find/parser"
+	"github.com/puppetlabs/wash/cmd/internal/find/types"
+	cmdutil "github.com/puppetlabs/wash/cmd/util"
+	"github.com/puppetlabs/wash/plugin"
+)
+
+type walker struct {
+	p    types.Predicate
+	opts types.Options
+	conn *client.DomainSocketClient
+}
+
+func newWalker(r parser.Result, conn *client.DomainSocketClient) *walker {
+	return &walker{
+		p:    r.Predicate,
+		opts: r.Options,
+		conn: conn,
+	}
+}
+
+func (w *walker) Walk(e types.Entry) {
+	w.walk(e, 0)
+}
+
+func (w *walker) walk(e types.Entry, depth uint) {
```

Comment:
Go's stack limit is 1 GB. On my machine, https://gist.github.com/ekinanp/c79b4d164665d23057e65496785cf816 ran for 5592386 iterations. I don't think we're going to have directory hierarchies that are that large, so I'd rather keep the (simple) recursive algorithm if I can. Happy to switch to a stack, but it'll be a bit more complicated.